src/tree/AttachedAccountsTreeItem.ts

```diff
@@ -32,12 +32,27 @@ export class AttachedAccountsTreeItem implements IAzureParentTreeItem {
     public childTypeLabel: string = 'Account';
 
     private readonly _serviceName = "ms-azuretools.vscode-cosmosdb.connectionStrings";
-    private _attachedAccounts: IAzureTreeItem[] = [];
+    private _attachedAccounts: IAzureTreeItem[] | undefined;
     private _keytar: typeof keytarType;
 
+    private _loadPersistedAccountsTask: Promise<IAzureTreeItem[]>;
+
     constructor(private readonly _globalState: vscode.Memento) {
         this._keytar = fetchNodeModule('keytar');
-        this.loadPersistedServers();
+        this._loadPersistedAccountsTask = this.loadPersistedAccounts();
+    }
+
+    private async getAttachedAccounts(): Promise<IAzureTreeItem[]> {
```

Comment:
Nit: Can we move this function to below loadMoreChildren? :P 
If I understand correctly, that's the only call site, and that makes it easier to read. 
Feel free to ignore.

Reply:
It's also called in 'attachAccount' and 'detach'